Source/ChannelsVRC7.cpp

```diff
@@ -235,14 +235,15 @@ unsigned int CChannelHandlerVRC7::GetFnum(int Note) const
 
 int CChannelHandlerVRC7::CalculateVolume() const
 {
-	int Volume = (m_iVolume >> VOL_COLUMN_SHIFT) - GetTremolo();
+	int Volume = (m_iInstVolume * (m_iVolume >> VOL_COLUMN_SHIFT)) / 15 - GetTremolo();		// // //
 	if (Volume > 15)
 		Volume = 15;
 	if (Volume < 0)
 		Volume = 0;
 	return Volume;		// // //
 }
```

Comment:
I replaced the function with `return LimitVolume((m_iVolume >> VOL_COLUMN_SHIFT) + m_iInstVolume - 15 - GetTremolo());` copying `CChannelHandlerS5B::CalculateVolume()`. The oddity is that `CChannelHandler::LimitVolume()` depends on *program settings* (`bCutVolume` -> "Cut sub-volume")for whether to clamp volumes quieter than 1 to 0, or leave them at 1. I'm not sure which one matches hardware behavior on 5B, and which one to use here, and which one to use for the NSF driver.

I think we should remove the "Cut sub-volume" option entirely (since it causes the same module to play differently for different users), and pick whichever one is hardware-accurate.

Reply:
You'll have to decide how the NSF driver should handle channel volume 5 + instrument volume 5, whether to play the note at volume 1, volume 0 (unsure if silent or not), or not at all. See #107.